queries/html/injections.scm

```diff
@@ -25,3 +25,112 @@
   (text) @injection.content
   (#eq? @_py_config "py-config")
   (#set! injection.language "toml"))
+
+;------ Angular queries ------
+
+; {{ someBinding }}
+((text) @injection.content 
+  (#lua-match? @injection.content "%{%{.+%}%}")
+  (#set! injection.language "angular")
+)
+
+; }
+((text) @injection.content 
+  (#lua-match? @injection.content "%}")
+  (#set! injection.language "angular")
+)
+
+; Angular control flow statements
+; @if (condition) {
+((text) @injection.content 
+  (#lua-match? @injection.content "%@if%s+%(.+%)%s+%{$")
+  (#set! injection.language "angular")
+)
+
+; @for (item of items) {
+((text) @injection.content 
+  (#lua-match? @injection.content "%@for%s+%(.+%)%s+%{$")
+  (#set! injection.language "angular")
+)
+
+; @switch (condition) {
+((text) @injection.content 
+  (#lua-match? @injection.content "%@switch%s+%(.+%)%s+%{$")
+  (#set! injection.language "angular")
+)
+
+; @case (value) {
+((text) @injection.content 
+  (#lua-match? @injection.content "%@case%s+%(.+%)%s+%{$")
+  (#set! injection.language "angular")
+)
+
+; @default {
+((text) @injection.content 
+  (#lua-match? @injection.content "%@default%s+%{$")
+  (#set! injection.language "angular")
+)
+
+; @defer {
+((text) @injection.content 
+  (#lua-match? @injection.content "%@defer%s+%{$")
+  (#set! injection.language "angular")
+)
+
+; @defer (conditions) {
+((text) @injection.content 
+  (#lua-match? @injection.content "%@defer%s+%(.+%)%{$")
+  (#set! injection.language "angular")
+)
+
+; Angular continuation blocks
+; } @else {
+((text) @injection.content
+  (#lua-match? @injection.content "%}%s+%@else%s+%{$")
+  (#set! injection.language "angular")
+)
+
+; } @else if (condition) {
+((text) @injection.content
+  (#lua-match? @injection.content "%}%s+%@else%s+if%s+%(.+%)%s+%{$")
+  (#set! injection.language "angular")
+)
+
+; } @empty {
+((text) @injection.content 
+  (#lua-match? @injection.content "%}%s+%@empty%s+%{$")
+  (#set! injection.language "angular")
+)
+
+; } @placeholder {
+((text) @injection.content
+  (#lua-match? @injection.content "%}%s+%@placeholder%s+%{$")
+  (#set! injection.language "angular")
+)
+
+; @placeholder (minumum) {
+((text) @injection.content
+  (#lua-match? @injection.content "%}%s+%@placeholder%s+%(.+%)%{$")
+  (#set! injection.language "angular")
+)
+
+; } @loading {
+((text) @injection.content
+  (#lua-match? @injection.content "%}%s+%@loading%s+%{$")
+  (#set! injection.language "angular")
+)
+
+; } @loading (conditions) {
+((text) @injection.content
+  (#lua-match? @injection.content "%}%s+%@loading%s+%(.+%)%s+%{$")
+  (#set! injection.language "angular")
+)
+
+; } @error {
+((text) @injection.content
+  (#lua-match? @injection.content "%}%s+%@error%s+%{$")
```

Comment:
@dlvandenberg Will a pattern of `(@error|@loading|@placeholder|@empty|@default|@defer|@if|@else|@switch)` be decent enough to detect html angular files? (given you can only check for the first 10 lines)

Reply:
tldr: the way to detect angular is to let vscode do it :sob:

Reply:
The parser should either extend html (in which case it probably won't need any injections), or do it the [htmldjango](https://github.com/interdependence/tree-sitter-htmldjango/blob/8873e3df89f9ea1d33f6235e516b600009288557/grammar.js#L149-L150) way and let html be injected in the remaining content.

Reply:
Either way is fine; we just need a decision and documentation (even if just in in the PR description here) to point people to if they ask.

Reply:
Yeah, and a PR would surely be welcome to add to that for angular (if necessary, increasing the number of lines). It doesn't have to be foolproof and perfect; you can always override this via a plugin or (project-specific) config.

Reply:
> Filetype would be `htmlangular` then?

If someone wants to contribute the syntax (or at least extend html).

Reply:
(not me)

Also, for highlighting angular files correctly, let's wait for https://github.com/dlvandenberg/tree-sitter-angular-2 and then see what would be the best

Reply:
> > @dlvandenberg Will a pattern of `(@error|@loading|@placeholder|@empty|@default|@defer|@if|@else|@switch)` be decent enough to detect html angular files? (given you can only check for the first 10 lines)
> 
> There are other patterns like the `[attr]=""` or `*structuralDirective` or `@animation` or `{{template}}` that all indicate an Angular file and may appear anywhere in the file. Even worse the a bunch of these are not unique to Angular.
> 
> But even with that, maybe, until the first file with a license header comes around...
> 
> I think the only reliable way to know if the file is an Angular template is if a parent directory contains an `angular.json` file (though even that is not technically correct... It's an Angular Template file if it is referenced by an Angular component)
> 
> The Angular LSP does it that way...

In case of a Nx monorepo, you don’t have any `angular.json` files. I’m not sure how to handle file detection yet. 

But the parser that I’m rewriting is extending the HTML one and assumes that the file is detected as an Angular template file.